trin-core/src/cli.rs

```diff
@@ -184,20 +191,41 @@ impl TrinConfig {
                 DEFAULT_WEB3_HTTP_ADDRESS => {}
                 _ => panic!("Must not supply an http address when using ipc protocol for json-rpc"),
             },
-            val => panic!("Unsupported json-rpc protocol: {}", val),
+            val => panic!("Unsupported json-rpc protocol: {val}"),
         }
 
-        match config.geth_url {
+        match config.pandaops_url {
```

Comment:
I haven't done a thorough review of the logic here, but it seems to me like some of this logic can be handled by [clap](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.requires_if) helpers.

Reply:
I really like the concept but ran into some trouble and so reverted the changes after not finding a solution. 
## Tests that should pass, fail
When I replaced the manual checks with clap helpers, I noticed that the unit tests that require a `should_panic` were not capturing the panic that clap triggers. 
1. A test had an incorrect flag combination 
2. Triggered a clap helper requirement 
3. Code panics (good)
4. Test (`should_panic`) does not pass (bad)

It seems that Clap triggers an early exit before the test can properly assess things. I also tried a `panic::catch_unwind(|| {})` based attempt, but no result. 
## Opacity during triggering
A separate but also important thing is that the panic messages Clap produced were without context and hard to determine where they came from (e.g,. because tests are run multithreaded, it would dump a the following (out of order, making it appear to be from a different (passing) test. 

```sh
test cli::test::test_http_protocol_rejects_custom_web3_ipc_path - should panic ... ok
error: The following required arguments were not provided:
    --local-provider-url <local-provider-url>

USAGE:
    trin --discovery-port <discovery-port> --kb <kb> --local-provider-url <local-provider-url> --master-accumulator-path <master-acc-path> --networks <networks>... --pandaops-url <pandaops-url> --pool-size <pool-size> --trusted-provider <trusted-provider> --web3-http-address <web3-http-address> --web3-ipc-path <web3-ipc-path> --web3-transport <web3-transport>
For more information try --helptest cli::test::test_ipc_with_custom_path ... ok
```
Running the test in single-thread mode allows the ordering to be correct. The error messages are still quite terse relative to the custom message, and are less useful to the end user.
## Outcome
I have kept the conditionals-based logic that I had previously